xbmc/addons/interfaces/General.cpp

```diff
@@ -413,6 +412,18 @@ int Interface_General::get_global_idle_time(void* kodiBase)
   return g_application.GlobalIdleTime();
 }
 
+char* Interface_General::get_current_skin_name(void* kodiBase)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (addon == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_General::%s - invalid data (addon='%p')", __FUNCTION__, addon);
+    return nullptr;
+  }
+
+  return strdup(CServiceBroker::GetSettings().GetString(CSettings::SETTING_LOOKANDFEEL_SKIN).c_str());
```

Comment:
The setting stores the skin id, not the name. I'm guessing you actually want is the id; if so the methods should say 'id'

Reply:
Is changed to use id instead of name